### ISSUE TITLE
Dock for light warships/small freighters/interceptors

### ISSUE THREAD

**Issue by smileyill** [endless-sky/endless-sky#4376]
So I've played about 200 hours of Endless Sky and it is a great game. I played a lot of Sins of a Solar Empire, but Endless Sky is better.

Why not create docking bays for small freighters/light warships/interceptors on specific larger ships? In Star Wars Corellian Corvettes can fit inside Star Destroyers. It would reduce CPU drag when moving large fleets and make capturing damaged ships easier to protect.

The large ships would need to use outfit space and cargo space. Perhaps ~20% more cargo space than the size of the dock plus that total in outfit space. Balanced well, the option would be well worth it. 

The docked ships could be deployed/recalled with "d" just like fighters, but with a ~200% longer delay.

**Comment by Amazinite** [comment]
Ref: https://github.com/endless-sky/endless-sky/pull/2200#issuecomment-283503934, https://github.com/endless-sky/endless-sky/issues/4033

**Comment by smileyill** [comment]
I see #2200 and #4033 mainly as discussions for using bays for better fighters whereas I am more interested in using bays to transport ships instead of them using their own hyper/jump/scram drives. For 3 main reasons, firstly to protect them on initial jumps into hostile systems, secondly for docking damaged ships for repairs, and thirdly for reducing CPU drain when moving 200 plus ships.

It would also allow for more intricate fleet maneuvers when combined with dividing of ships in different groups with ctrl-1,2,3 etc. For instance, light warship group 1 could be deployed before a jump and placed on "hold position" while 2 groups of heavy warships jump and engage the enemy fleet. Heavy warship group 1 could have the docking bays and be positioned to tank and when damaged retreated behind group 2 of more combat outfitted/type heavy warships and then light warship group 1 jumped in to support for a short time and then retreated to heavy warship group 1 and docked for repairs or even reloading. Of course, that is a simple maneuver compared to what would be possible.

**Comment by petervdmeer** [comment]
The current code uses a boolean to differentiate between bay types, effectively limiting the amount of possible bay-types to 2.

Change #4391 makes the code for bays more generic, which is the first step in allowing more that 2 bay types.

Some other changes are also needed before we can consider more bay types. Most notable is that a player is currently not allowed to fly around in a ship that can be carried. We will need to do something with this type of exceptions before we can allow more bay types.

**Comment by Pointedstick** [comment]
In principle I don't see anything wrong with this request, and I can see it being used particularly for modders. But for the base game, most ships in Endless Sky aren't large enough to feasibly contain even your average light warship.

**Comment by MCOfficer** [comment]
it would be nice for ES's data format to be so modular that everything can have (almost) any attribute. admittedly, that would produce some very quirky results at times, but vanilla doesn't have to use them.

**Comment by Zitchas** [comment]
That's true: Only the biggest ships (Bactrian and World Ship in particular) would be big enough to fit larger ships out of the current vanilla ships, but there are plug-ins out there that expand the ship repertoire into lots of larger ships for which this would be useful.

Quirky can be fun (so long as it mostly insn't vanilla).

**Comment by Amazinite** [comment]
>Most notable is that a player is currently not allowed to fly around in a ship that can be carried. We will need to do something with this type of exceptions before we can allow more bay types.

Would #4391 allow for the easy addition of new carried-only categories, though? e.g. how [I've suggested](https://github.com/endless-sky/endless-sky/pull/2200#issuecomment-283503934) a category above fighter in terms of weight of carried ships be named gunships.

**Comment by petervdmeer** [comment]
> Would #4391 allow for the easy addition of new carried-only categories, though? e.g. how [I've suggested](https://github.com/endless-sky/endless-sky/pull/2200#issuecomment-283503934) a category above fighter in terms of weight of carried ships be named gunships.

Yes

**Comment by Amazinite** [comment]
In v0.9.15 with #5971 we made it so that which ships can be carried is configurable in the data instead of being hardcoded to just fighters and drones. If a plugin wishes to have larger ships that can be docked then they can do so, but for the vanilla game I don't think it makes sense to have these ship categories dock.
